enhancements/kubelet/node-swap.md

```diff
@@ -0,0 +1,310 @@
+be able to use the feature correctly.
+
+Therefore, a user of OpenShift version >4.9.0 can enable the feature flag, but
+they also will require a version of CRI-O that supports the
+`MemorySwapLimitInBytes` for best results. This is only supported in the 1.23
+and onwards releases of CRI-O, and should be supported in the first release of
+OpenShift 4.10.
+
+### Configure worker Kubelets
+
+Swap behaviour on a node can be configured with
+[`memorySwap.SwapBehavior`](https://kubernetes.io/docs/concepts/architecture/nodes/#swap-memory).
+
+The most straightforward way to configure the kubelets with this feature is
+with a custom `KubeletConfig` that will automatically be applied by the MCO:
+
+```bash
+# Enable the custom kubelet config on the worker pool
+oc label machineconfigpool worker custom-kubelet=enabled
+```
+
+```yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: KubeletConfig
+metadata:
+  name: custom-config 
+spec:
+  machineConfigPoolSelector:
+    matchLabels:
+      custom-kubelet: enabled 
+  kubeletConfig: 
+    failSwapOn: false
+    memorySwap:
+      swapBehavior: UnlimitedSwap  # LimitedSwap is also supported
+```
+
+Note that enabling swap on control plane nodes is possible but **not** recommended.
+
+### Add swap to nodes
+
+There are a few different ways this can be accomplished. The most
+straightforward is to add a kubelet configuration that enables a swapfile at
+startup with a custom machine config, the same way we enable swap in upstream
+CI:
+
+```yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfig
+metadata:
+  labels:
+    machineconfiguration.openshift.io/role: worker
+  name: 90-worker-swap
+spec:
+  config:
+    ignition:
+      version: 3.2.0
+    systemd:
+      units:
+      - contents: |
+          [Unit]
+          Description=Enable swap on CoreOS
+          Before=crio-install.service
+          ConditionFirstBoot=no
+          
+          [Service]
+          Type=oneshot
+          ExecStart=/bin/sh -c "sudo dd if=/dev/zero of=/var/swapfile count=1024 bs=1MiB && sudo chmod 600 /var/swapfile && sudo mkswap /var/swapfile && sudo swapon /var/swapfile && free -h"
+          [Install]
+          
+          WantedBy=multi-user.target
+        enabled: true
+        name: swap-enable.service
+```
+
+It is also possible to use [ignition
+configs](https://coreos.github.io/ignition/configuration-v3_3/) to add swap
+partitions to worker nodes with `filesystems.format = swap`.
+
+Beyond tech preview, we may want to look into adding swap support to the
+installer, and consider adding a partition on the root node, or perhaps
+[provision and mount an NVMe
+volume](https://github.com/openshift/machine-config-operator/issues/1619).
+
+### Open Questions [optional]
+
+- Will we eventually want to enable swap on all OpenShift nodes by default?
+- Should swap just be limited to worker nodes, or should we consider adding it
+  to control plane nodes too?
```

Comment:
Could add a line saying that our testing will help answer these questions.

Reply:
Done!